packages/transformers/assemblyscript/src/ASTransformer.js

```diff
@@ -0,0 +1,144 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+import {promises as fsp} from 'fs';
+import path from 'path';
+import {tmpdir} from 'os';
+import {Transform} from 'stream';
+
+const MARKER = 'asc:';
+const PREFIX_MATCHER = /^asc:(.+)$/;
+
+// This special import contains the `compileStreaming` polyfill.
+const SPECIAL_IMPORT = '__rollup-plugin-assemblyscript_compileStreaming';
+
+const compileStreamingCode = `
+  export async function compileStreaming(respP) {
+              if('compileStreaming' in WebAssembly) {
+                return WebAssembly.compileStreaming(respP);
+              }
+              return respP
+                .then(resp => resp.arrayBuffer())
+                .then(buffer => WebAssembly.compile(buffer));
+            }
+        `;
+
+export default (new Transformer({
+  async transform({asset, options}) {
+    const asCompiler = await getASCompilerPath(asset.filePath, options);
```

Comment:
Looks this like this was taken from the Elm transformer? This workaround was needed there because the `elm` package has no JS entry and so `require("elm")` always fails.

I think you can instead simply do 
```js
let asCompiler  = await options.packageManager.require('assemblyscript/cli/asc', asset.filePath, {
	shouldAutoInstall: options.shouldAutoInstall,
}),
```
here (or whatever subpath needs to be imported)

Reply:
Fixed